src/Api/Clients/ProductClient.cs

```diff
@@ -10,39 +12,94 @@ public interface IProductClient
         Task<ApiResponse<IEnumerable<Product>>> GetProductsAsync();
         Task<ApiResponse<ProductTicker>> GetProductTickerAsync(string productId);
         Task<ApiResponse<OrderBook>> GetOrderBookAsync(string productId, int level = 1);
-    }
+		Task<List<Candle>> GetHistoricRatesAsync(string i_product_id, DateTimeOffset i_start, DateTimeOffset i_end, TimeSpan i_granularity);
+	}
 
-    public class ProductClient : GdaxClient, IProductClient
-    {
-        public ProductClient(string baseUrl, RequestAuthenticator authenticator)
-            : base(baseUrl, authenticator)
-        {
-        }
-
-        public async Task<ApiResponse<IEnumerable<Product>>> GetProductsAsync()
-        {
-            return await this.GetResponseAsync<IEnumerable<Product>>(
-                new ApiRequest(HttpMethod.Get, "/products")
-            );
-        }
-
-        public async Task<ApiResponse<ProductTicker>> GetProductTickerAsync(string productId)
-        {
-            var response = await this.GetResponseAsync<ProductTicker>(
-                new ApiRequest(HttpMethod.Get, $"/products/{productId}/ticker")
-            );
-
-            if(response.Value != null)
-                response.Value.product_id = productId;
-
-            return response;
-        }
-
-        public async Task<ApiResponse<OrderBook>> GetOrderBookAsync(string productId, int level = 1)
-        {
-            return await this.GetResponseAsync<OrderBook>(
-                new ApiRequest(HttpMethod.Get, $"/products/{productId}/book?level={level}")
-            );
-        }
-    }
+	public class ProductClient : GdaxClient, IProductClient
```

Comment:
The indenting problem causes this to be regarded as changed code, even though it is not. You probably should resubmit the PR with the indenting problem fixed.